### PR TITLE
feat: Add fog-of-war circuit + core commit contract call

### PR DESCRIPTION
**Pull request by nalinbhardwaj**

### CODE REVIEW

File: circuits/commiter.circom [hunk 1]
```diff
@@ -0,0 +1,89 @@
+pragma circom 2.0.3;
+
+include "../node_modules/circomlib/circuits/poseidon.circom";
+include "../node_modules/circomlib/circuits/comparators.circom";
+include "../node_modules/circomlib/circuits/gates.circom";
+include "../node_modules/circomlib/circuits/mimcsponge.circom";
+// include "merkle.circom";
+// include "https://github.com/0xPARC/circom-secp256k1/blob/master/circuits/bigint.circom";
```
Comment (phated) [inline circuits/commiter.circom]:
```suggestion
```

File: contracts/facet/CoreFacet.sol [hunk 2]
```diff
@@ -2,9 +2,40 @@
 pragma solidity ^0.8.0;
 
 import {WithStorage} from "../library/LibStorage.sol";
+import {LibMIMC} from "../library/LibMIMC.sol";
+import {Verifier as CommiterVerifier} from "../library/CommiterVerifier.sol";
+import "hardhat/console.sol";
```
Comment (phated) [inline contracts/facet/CoreFacet.sol]:
```suggestion
```

File: contracts/facet/CoreFacet.sol [hunk 3]
```diff
@@ -2,9 +2,40 @@
...
 
 contract CoreFacet is WithStorage {
     constructor() {}
 
     function action() public {}
```
Comment (phated) [inline contracts/facet/CoreFacet.sol]:
```suggestion
```

File: contracts/library/LibMIMC.sol [hunk 4]
```diff
@@ -0,0 +1,291 @@
...
+        return currentState;
+    }
+
+    function mimcSponge(uint256[] memory inputs, uint256 outputCount, uint256 rounds, uint256 key) internal pure returns (uint256[] memory) {
+        uint256[220] memory roundConstants = getRoundConstants(); 
+        FeistelState memory state = FeistelState(0, 0);
+        for (uint256 i = 0;i < inputs.length;i++) {
+            state = inject(state, (inputs[i] % p));
+            // console.log('stateL injected', state.l);
+            // console.log('stateR injected', state.r);
```
Comment (phated) [inline contracts/library/LibMIMC.sol]:
```suggestion
```

File: contracts/library/LibMIMC.sol [hunk 5]
```diff
@@ -0,0 +1,291 @@
...
+            state = mix(state, rounds, key, roundConstants);
+        }
+        // console.log('stateL', state.l);
+        // console.log('stateR', state.r);
```
Comment (phated) [inline contracts/library/LibMIMC.sol]:
```suggestion
```

File: contracts/library/LibStorage.sol [hunk 6]
```diff
@@ -1,12 +1,19 @@
...
+}
+
 struct GameStorage {
     // Contract housekeeping
     address diamondAddress;
     // Admin controls
     bool paused;
     // Game world state
+    uint256 saltUpperBound;
+    uint256 gridUpperBound;
```
Comment (phated) [inline contracts/library/LibStorage.sol]:
```suggestion
    // saltUpperBound might move to the PlayerState in the future
    uint256 saltUpperBound;
    uint256 gridUpperBound;
```

**Review by phated — COMMENTED** [review]
Looking great! A few code cleanup things and we also chatted about adding `saltUpperBound` and `gridUpperBound` in the zkgame.toml and set them in the initializer contract

**Comment by nalinbhardwaj** [comment]
> adding saltUpperBound and gridUpperBound in the zkgame.toml and set them in the initializer contract

done

**Review by phated — APPROVED** [review]

**Pull request merged**